iree/compiler/Dialect/HAL/Target/SPIRVCommon/SPIRVTarget.cpp

```diff
@@ -211,7 +209,7 @@ LogicalResult SPIRVTargetBackend::recordDispatch(
           spvFuncOp.getParentOfType<ModuleOp>(), numWorkgroupsFnAttr));
       if (!numWorkgroupsFn) return failure();
       workgroupCount = calculateWorkgroupCountFromNumWorkgroupsFn(
-          loc, numWorkgroupsFn, executableOp.getInterfaceOp(),
+          loc, numWorkgroupsFn, dispatchState.executableOp.getInterfaceOp(),
```

Comment:
(not needed here but could happen in a followup) if we passed the interface in then we could potentially remove executableOp from the dispatch state entirely